emulator/src/test/java/com/scalar/client/tool/emulator/EmulatorTerminalTest.java

```diff
@@ -91,4 +91,109 @@ public void input_withSpacesForFlagInput_ShouldReturnProperParamsArrayProperly()
     // Assert
     Assert.assertArrayEquals(paramsArray, mockedParamsArray);
   }
+
+  @Test
+  public void get_withSpacesForInput_ShouldReturnProperParamsArrayProperly() {
+    // Arrange
+    String input = "get \"key with space\"";
+    String[] mockedParamsArray = { "get", "key with space" };
+
+    // Action
+    String[] paramsArray = emulatorTerminal.paramsParser(input);
+
+    // Assert
+    Assert.assertArrayEquals(paramsArray, mockedParamsArray);
+  }
+
+  @Test
+  public void getJ_withSpacesForInput_ShouldReturnProperParamsArrayProperly() {
+    // Arrange
+    String input = "get \"key with space\"";
+    String[] mockedParamsArray = { "get", "key with space" };
+
+    // Action
+    String[] paramsArray = emulatorTerminal.paramsParser(input);
+
+    // Assert
+    Assert.assertArrayEquals(paramsArray, mockedParamsArray);
+  }
+
+  @Test
+  public void get_curlyBracket_ShouldReturnProperParamsArrayProperly() {
+    // Arrange
+    String input = "get { \"asset_id \": \"key with space\"}";
+    String[] mockedParamsArray = { "get", "{ \"asset_id \": \"key with space\"}" };
+
+    // Action
+    String[] paramsArray = emulatorTerminal.paramsParser(input);
+
+    // Assert
+    Assert.assertArrayEquals(paramsArray, mockedParamsArray);
+  }
+
+  @Test
+  public void scan_withSpacesForInput_ShouldReturnProperParamsArrayProperly() {
+    // Arrange
+    String input = "scan \"key with space\"";
+    String[] mockedParamsArray = { "scan", "key with space" };
+
+    // Action
+    String[] paramsArray = emulatorTerminal.paramsParser(input);
+
+    // Assert
+    Assert.assertArrayEquals(paramsArray, mockedParamsArray);
+  }
+
+  @Test
+  public void scanJ_withSpacesForInput_ShouldReturnProperParamsArrayProperly() {
+    // Arrange
+    String input = "scan -j \"key with space\"";
+    String[] mockedParamsArray = { "scan", "-j", "key with space" };
+
+    // Action
+    String[] paramsArray = emulatorTerminal.paramsParser(input);
+
+    // Assert
+    Assert.assertArrayEquals(paramsArray, mockedParamsArray);
+  }
+
+  @Test
+  public void scan_curlyBracket_ShouldReturnProperParamsArrayProperly() {
```

Comment:
Ditto

Reply:
Thank you for detailed review.
The test case method name updated.